ddl/ddl_api.go

```diff
@@ -5761,6 +5762,11 @@ func buildPlacementSpecs(bundle *placement.Bundle, specs []*ast.PlacementSpec) (
 
 			// alter == drop + add new rules
 			if spec.Tp == ast.PlacementDrop {
+				// error if no rules will be dropped
+				if len(bundle.Rules) == origLen {
+					err = errors.Errorf("there is no rule to drop")
```

Comment:
The pattern of the error message should be like "ErrInvalidPlacementSpec".  Calling `Errorf` directly doesn't obey the rule.

Reply:
It will be wraped by `ErrInvalidPlacementSpec`at L5785.

Reply:
OK